### ISSUE TITLE
Expose Given values in Runner API

### ISSUE THREAD

**Issue by rtfeldman** [project-fuzzball/test#11]
Currently when there are "Given" values, we prepend them to the failure strings.

It would be better to expose them distinct from the failure string. That way different runners could format them differently.

However, this is a complicated change. In particular it means that `Expect.getFailure` needs to return something richer than a `String`, and `Expect.all` needs to change its internal implementation.

**Comment by mgold** [comment]
Would the input values still be strings? If so a tuple or record would work. If not, I'm not sure how the polymorphism would work.

**Comment by rtfeldman** [comment]
Yeah, strings was what I had in mind.

**Comment by mgold** [comment]
Okay, sounds fine since they're only going to be spit out as strings anyway. (i.e. not asking Elm code to work with them) So, what sort of output would we like runners to have that they can't have right now? (Use-case driven design.)

**Comment by rtfeldman** [comment]
Specifically around styling the "Given" context (the values the Fuzzer generated) so it's clearly distinct at a glance from the failure messages themselves. I'd like to render them gray, like so:

![screen shot 2016-06-28 at 3 06 13 pm](https://cloud.githubusercontent.com/assets/1094080/16433988/eab0925c-3d41-11e6-815e-780c3874830a.png)

Unfortunately, this requires running different code in a HTML runner versus in a CLI runner. That means my options are:
- Run a regular expression to parse out "Given" and then decorate as appropriate. (Messy, slow, prone to false positives if end user's input includes what I'm looking for.)
- Semantically separate the "Given" from the failure message, so each runner can style them separately. (What I'd like to do.)
- Just leave it at one big String and make do without this formatting. (What I'm currently doing.)

Currently I'm making an ASCII art "underline" that doesn't look great in either runner, but which seems "good enough" for now:

![screen shot 2016-06-28 at 3 03 19 pm](https://cloud.githubusercontent.com/assets/1094080/16433908/84c086a0-3d41-11e6-9b37-6ea35492aafa.png)

tl;dr I think the top screenshot looks a lot better, and is worth aiming for. 😄

**Comment by mgold** [comment]
I liked the inward triangle, but yeah the top screenshot looks good.

**Comment by rtfeldman** [comment]
![screen shot 2016-06-29 at 3 19 06 pm](https://cloud.githubusercontent.com/assets/1094080/16470757/e9eb2e92-3e0c-11e6-99ca-d94b8709eda7.png)

![screen shot 2016-06-29 at 3 20 05 pm](https://cloud.githubusercontent.com/assets/1094080/16470768/fbdc1c56-3e0c-11e6-9b28-3fc2b6ad3e68.png)
